File: internal/templates/templates/todo-api/_python/services/list-todos/handler.py
"""GET /todos — every todo in the table."""

import json
import os

import boto3

table = boto3.resource("dynamodb").Table(os.environ["TABLE_NAME"])


def handler(event, context):
    items = table.scan().get("Items", [])
    # default=str: DynamoDB numbers come back as Decimal
    return {"statusCode": 200, "body": json.dumps(items, default=str)}

File: internal/templates/templates/todo-api/_node/services/list-todos/handler.mjs
// GET /todos — every todo in the table.
import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import { DynamoDBDocumentClient, ScanCommand } from "@aws-sdk/lib-dynamodb";

const db = DynamoDBDocumentClient.from(new DynamoDBClient({}));

export const handler = async () => {
  const { Items = [] } = await db.send(new ScanCommand({ TableName: process.env.TABLE_NAME }));
  return { statusCode: 200, body: JSON.stringify(Items) };
};
